### ISSUE TITLE
Create unit tests

### ISSUE THREAD

**Issue by adrcunha** [knative/docs#66]
"unit tests" in this context is not restricted to code, might even include linting markdown files.

**Comment by samodell** [comment]
@adrcunha  Would you consider our current testing sufficient to close out this issue? Between the checks built in to Prow and @mattmoor 's Sockpuppet bot, I feel pretty good about our coverage.

**Comment by adrcunha** [comment]
TL;DR: No.

Currently there are no unit tests. At all:

https://github.com/knative/docs/blob/df75e7e4885915bdb17f2a88046c79aa0d0d468a/test/presubmit-tests.sh#L28

Also there are no build tests, and the E2E tests are just checking that Knative Serving starts (which is not affected by any changes in this repo).

Currently there's nothing preventing `doc` from getting broken code merged (e.g., samples), or even bad markdown files (I dunno how often Matt's bot lints/validates/fixes them). Although some of the infrastructure required is in place (e.g., https://github.com/knative/test-infra/pull/192), it's not active yet.

**Comment by samodell** [comment]
Thanks for the background. Is there any kind of timeline for this work yet?

**Comment by adrcunha** [comment]
Not that I am aware of.  But we can definitely bring this up in the next documentation or productivity meeting.

**Comment by samodell** [comment]
That would be great! I'll add you to the agenda for the documentation working group on 1/8.

**Comment by adrcunha** [comment]
It seems that the amount of testing we currently have is enough.

/close
